### DISCUSSION TITLE
Add support for `INotifyPropertyChanging`

### DISCUSSION THREAD

**Discussion by Cologler** [canton7/PropertyChanged.SourceGenerator#13] [Ideas] [answered]
Since EF core [required](https://docs.microsoft.com/en-us/ef/core/change-tracking/change-detection) `INotifyPropertyChanging`, if a type is implemented `INotifyPropertyChanging`, it should also fire the `PropertyChanging` event.

**Comment by canton7** [comment]
Fair request. It opens up a bit of a can of worms around whether we support `On{PropertyName}Changing`, `OnAnyPropertyChanging`, etc. We'll also need to duplicate all of the logic to find whether the base class already implements this interface, find an appropriate base class method to call, etc. There's also questions around whether such methods would be passed the old and new values of the property, in the same way that they do for `PropertyChanged`.

There's also the question of whether we automatically add `INotifyPropertyChanging`, or whether we get some indication from the user that they want us to implement this interface.

It looks like Fody went the whole hog and implemented all of the above (although they only allow handlers to receive the "before" value). It's a bit of a refactor for PropertyChanged.SourceGenerator, but hopefully not to bad.

**Reply by canton7** [reply]
Opened #14

**Reply by canton7** [reply]
Apologies for the delay -- this took a while to implement. I think it's pretty much there though.

**Comment by canton7** [comment, marked as answer]
Released in 1.0.4
